homeassistant/components/device_tracker/unifi_direct.py

```diff
@@ -0,0 +1,184 @@
+"""
+import logging
+import json
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.device_tracker import (
+    DOMAIN, PLATFORM_SCHEMA, DeviceScanner,
+    load_config, CONF_TRACK_NEW, YAML_DEVICES)
+from homeassistant.const import (
+    CONF_HOST, CONF_PASSWORD, CONF_USERNAME,
+    CONF_PORT)
+
+REQUIREMENTS = ['pexpect==4.0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_SSH_PORT = 22
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_SSH_PORT): cv.port
+})
+
+
+# pylint: disable=unused-argument
+def get_scanner(hass, config):
+    """Validate the configuration and return a Unifi direct scanner."""
+    try:
+        return UnifiDeviceScanner(config[DOMAIN], hass)
+    except (ConnectionError, KeyError):
+        return False
+
+
+class UnifiDeviceScanner(DeviceScanner):
+    """This class queries Unifi wireless access point."""
+
+    def __init__(self, config, hass):
+        """Initialize the scanner."""
+        self.host = config[CONF_HOST]
+        self.username = config[CONF_USERNAME]
+        self.password = config[CONF_PASSWORD]
+        self.port = config[CONF_PORT]
+        self.ssh_client = SshConnection(self.host, self.port,
+                                        self.username, self.password)
+        self.last_results = {}
+        self.devs_to_track = []
+
+        self.track_new = config.get(CONF_TRACK_NEW)
+        yaml_path = hass.config.path(YAML_DEVICES)
+        known_devices = load_config(yaml_path, hass, 0)
+
+        if known_devices:
+            for device in known_devices:
+                if device.mac:
+                    if device.track:
+                        _LOGGER.debug("Adding %s to Unifi direct tracker",
+                                      device.mac)
+                        self.devs_to_track.append(device.mac)
+
+    def scan_devices(self):
+        """Scan for new devices and return a list with found device IDs."""
+        return self._update_info()
+
+    def get_device_name(self, device):
+        """Return the name of the given device or None if we don't know."""
+        if not self.last_results:
+            return None
+        for key, value in self.last_results.items():
+            if key.upper() == device.upper():
+                hostname = value.get('hostname')
+                if hostname is not None:
+                    return str(hostname)
+                else:
+                    return None
+        return None
+
+    def _update_info(self):
+        """Ensure the information from the Unifi AP is up to date."""
+        self.last_results = self.ssh_client.get_result()
+        client_macs = []
+        for client in self.last_results.keys():
+            if client.upper() in self.devs_to_track or self.track_new:
```

Comment:
This filtering is handled by the device_tracker base component. Why do you need to do it here too?

Reply:
Didn't seem to work without it. I'll try removing it and see what happens, but when I first started this setting track_new to false had no effect, all devices showed up.

Reply:
Devices will still be added to known_devices.yaml, but they will be set to not track.

Reply:
I removed this check and it isn't working. All the devices get added and they are all set with `track: true`

Reply:
In your test, is this platform the first platform under device_tracker? It's only the configuration of the first platform in the list of platforms that sets the settings for the component.

Reply:
Oooo no it isn't I have two other trackers setup. That's kinda confusing. I'll try that.

Reply:
Yes, I agree, it's confusing.

Reply:
Okay, so that works. Now the question is, is that how it should be? Should something be done to allow that setting to work at the platform level? I think that would eliminate the confusion. A setting under platform shouldn't effect another platform in my opinion.

Reply:
Yes, I think so, but that's a refactor for another PR. An idea is to create one `DeviceTracker` instance per platform. Create the dictionary of devices first, check for uniqueness, and pass the same dictionary to all instances. I haven't checked all consequences yet.

Reply:
Cool. Thanks for the help. I'll push these changes in a minute and see what the tests do...